overlord/overlord.go

```diff
@@ -403,7 +403,11 @@ func (o *Overlord) Loop() {
 			case <-o.pruneTicker.C:
 				st := o.State()
 				st.Lock()
-				st.Prune(pruneWait, abortWait, pruneMaxChanges)
+				// account for deviceMgr == nil as it's not always present in
+				// tests.
+				if o.deviceMgr == nil || o.deviceMgr.CanPrune() {
+					st.Prune(pruneWait, abortWait, pruneMaxChanges)
```

Comment:
Note, this change, in case of multiple failures on seeding, will result in growing state (failed "seed" changes will not get pruned until one finally succeeds, which may never happen because if it fails, it probably fails for good), I wonder if this is a valid concern. I may need to rethink this. Perhaps a better fix is inside state.Purge(), where the logic around max number of changes could always be executed. CC @pedronis